application/src/main/java/org/gxf/soapbridge/soap/endpoints/SoapEndpoint.java

```diff
@@ -0,0 +1,266 @@
+import java.util.*;
+import java.util.stream.Collectors;
+import org.gxf.soapbridge.application.services.ConnectionCacheService;
+import org.gxf.soapbridge.application.services.SigningService;
+import org.gxf.soapbridge.configuration.properties.SoapConfigurationProperties;
+import org.gxf.soapbridge.kafka.senders.ProxyRequestKafkaSender;
+import org.gxf.soapbridge.soap.clients.Connection;
+import org.gxf.soapbridge.soap.exceptions.ConnectionNotFoundInCacheException;
+import org.gxf.soapbridge.soap.exceptions.ProxyServerException;
+import org.gxf.soapbridge.valueobjects.ProxyServerRequestMessage;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.stereotype.Component;
+import org.springframework.web.HttpRequestHandler;
+
+/**
+ * This {@link @Component} class is the endpoint for incoming SOAP requests from client
+ * applications.
+ */
+@Component
+public class SoapEndpoint implements HttpRequestHandler {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SoapEndpoint.class);
+
+  private static final String URL_PROXY_SERVER = "/proxy-server";
+
+  private static final int INVALID_CUSTOM_TIME_OUT = -1;
+
+  /** Service used to cache incoming connections from client applications. */
+  private final ConnectionCacheService connectionCacheService;
+
+  private final SoapConfigurationProperties soapConfiguration;
+
+  /** Message sender which can send a webapp request message to ActiveMQ. */
+  private final ProxyRequestKafkaSender proxyRequestsSender;
+
+  /** Service used to sign the content of a message. */
+  private final SigningService signingService;
+
+  /** Map of time-outs for specific functions. */
+  private final Map<String, Integer> customTimeOutsMap = new HashMap<>();
+
+  public SoapEndpoint(
+      final ConnectionCacheService connectionCacheService,
+      final SoapConfigurationProperties soapConfiguration,
+      final ProxyRequestKafkaSender proxyRequestsSender,
+      final SigningService signingService) {
+    this.connectionCacheService = connectionCacheService;
+    this.soapConfiguration = soapConfiguration;
+    this.proxyRequestsSender = proxyRequestsSender;
+    this.signingService = signingService;
+  }
+
+  @PostConstruct
+  public void init() {
+    final String[] split = soapConfiguration.getCustomTimeouts().split(",");
+    for (int i = 0; i < split.length; i += 2) {
+      final String key = split[i];
+      final Integer value = Integer.valueOf(split[i + 1]);
+      LOGGER.debug("Adding custom time out with key: {} and value: {}", key, value);
+      customTimeOutsMap.put(key, value);
+    }
+    LOGGER.debug("Added {} custom time outs to the map", customTimeOutsMap.size());
+  }
+
+  /** Handles incoming SOAP requests. */
+  @Override
+  public void handleRequest(
```

Comment:
Ik zou de logica wel heel graag meer uide de controller halen en meer in services zetten.